spec/features/saml/loa3_sso_spec.rb

```diff
@@ -33,6 +33,22 @@
     end
   end
 
+  context 'canceling verification', js: true do
+    it 'returns user to profile page if they have previously signed up' do
+      sign_in_and_2fa_user
+
+      loa3_sp_session do |session|
+        session['warden.user.user.session'][:first_time_recovery_code_view] = true
+      end
+
+      visit verify_path
+
+      click_on t('links.cancel')
+      click_on t('loa3.buttons.cancel')
+      expect(current_url).to match(/profile/)
```

Comment:
That is the intended behavior, if that key is missing, the user hasn't seen their recovery code, and is going to get redirected to the recovery code screen so they can write it down. If the session key is present, then they'll be redirected to their profile.

In verifying this manually, I did notice that the current behavior is to delete the user's account if they cancel identity verification when signing up. I think that redirecting to the recovery code page is superfluous for now, since there is no way for a user to bypass it.

If they sign up with either loa1 or 3, and complete the process, they'll see the recovery code, and if they complete loa3 on a separate session, they will have see the page as well.